src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -37,42 +38,44 @@ internal static class DependentTypeFinder
         /// <summary>
         /// Used for implementing the Inherited-By relation for progression.
         /// </summary>
-        internal static Task<IEnumerable<INamedTypeSymbol>> FindImmediatelyDerivedClassesAsync(
+        internal static Task<IEnumerable<SymbolAndProjectId<INamedTypeSymbol>>> FindImmediatelyDerivedClassesAsync(
             INamedTypeSymbol type,
             Solution solution,
             CancellationToken cancellationToken)
         {
-            return FindDerivedClassesAsync(type, solution, projects: null,
+            return FindDerivedClassesAsync(
+                SymbolAndProjectId.Create(type, projectId: null), solution, projects: null,
                 transitive: false, cancellationToken: cancellationToken);
         }
 
         /// <summary>
-        /// This is an internal implementation of <see cref="SymbolFinder.FindDerivedClassesAsync"/>, which is a publically callable method.
+        /// This is an internal implementation of <see cref="SymbolFinder.FindDerivedClassesAsync(SymbolAndProjectId{INamedTypeSymbol}, Solution, IImmutableSet{Project}, CancellationToken)"/>, which is a publically callable method.
         /// </summary>
-        public static Task<IEnumerable<INamedTypeSymbol>> FindTransitivelyDerivedClassesAsync(
+        public static Task<IEnumerable<SymbolAndProjectId<INamedTypeSymbol>>> FindTransitivelyDerivedClassesAsync(
             INamedTypeSymbol type,
             Solution solution,
             IImmutableSet<Project> projects,
             CancellationToken cancellationToken)
         {
-            return FindDerivedClassesAsync(type, solution, projects,
+            return FindDerivedClassesAsync(
+                SymbolAndProjectId.Create(type, projectId: null), solution, projects,
```

Comment:
Good question (and i'll doc as such): It effectively means: you have a symbol that we cannot serialize/deserialize across a process.  THat's because we won't know where to find teh symbol again on the other side.